### PR TITLE
First API Version reflecting a few endpoints

### PR DESCRIPTION
**Pull request by asilvis**
An initial version of API using only 3 endpoints:
- Incidents
- Components
- Component Groups

Those are using the same interface contract, so they do not implement methods outside a CRUD context.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  preset: 'ts-jest',
+  testEnvironment: 'node',
```
Comment (maveco69) [inline jest.config.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
  testEnvironment: "node",
```

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  preset: 'ts-jest',
```
Comment (maveco69) [inline jest.config.js:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
  preset: "ts-jest",
```

File: src/statuspage.test.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import DefaultOperations from './defaultOperations';
```
Comment (maveco69) [inline src/statuspage.test.ts:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
import DefaultOperations from "./defaultOperations";
```

File: src/statuspage.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import DefaultOperations from './defaultOperations';
```
Comment (maveco69) [inline src/statuspage.ts:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
import DefaultOperations from "./defaultOperations";
```

File: src/statuspage.test.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import StatusPage from './statuspage';
```
Comment (maveco69) [inline src/statuspage.test.ts:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
import StatusPage from "./statuspage";
```

File: src/statuspage.test.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+const pageId = "12345";
+const apiKey = "##key##";
+
+test('I can import StatusPage', () => {
+    expect(StatusPage).toBeDefined();
+});
+
+test('I can require StatusPage', () => {
```
Comment (maveco69) [inline src/statuspage.test.ts:11]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
test("I can require StatusPage", () => {
```

File: src/statuspage.test.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    expect(require('./StatusPage')).toBeDefined();
```
Comment (maveco69) [inline src/statuspage.test.ts:12]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
    expect(require("./StatusPage")).toBeDefined();
```

File: src/statuspage.test.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import DefaultOperations from './defaultOperations';
+import StatusPage from './statuspage';
+
+const pageId = "12345";
+const apiKey = "##key##";
+
+test('I can import StatusPage', () => {
```
Comment (maveco69) [inline src/statuspage.test.ts:7]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
test("I can import StatusPage", () => {
```

File: jest.config.js (hunk 1 above)
Comment (maveco69) [inline jest.config.js]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
  testEnvironment: "node",
```

File: src/statuspage.ts (hunk 4 above)
Comment (maveco69) [inline src/statuspage.ts:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
import DefaultOperations from "./defaultOperations";
```

File: jest.config.js (hunk 2 above)
Comment (maveco69) [inline jest.config.js:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
  preset: "ts-jest",
```

File: src/statuspage.test.ts (hunk 3 above)
Comment (maveco69) [inline src/statuspage.test.ts:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
import DefaultOperations from "./defaultOperations";
```

File: src/statuspage.test.ts (hunk 6 above)
Comment (maveco69) [inline src/statuspage.test.ts:11]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
test("I can require StatusPage", () => {
```

File: src/statuspage.test.ts (hunk 8 above)
Comment (maveco69) [inline src/statuspage.test.ts:7]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
test("I can import StatusPage", () => {
```

File: src/statuspage.test.ts (hunk 5 above)
Comment (maveco69) [inline src/statuspage.test.ts:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
import StatusPage from "./statuspage";
```

File: src/statuspage.test.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    expect(StatusPage).toBeDefined();
+});
+
+test('I can require StatusPage', () => {
+    expect(require('./statuspage')).toBeDefined();
```
Comment (maveco69) [inline src/statuspage.test.ts:12]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
    expect(require("./statuspage")).toBeDefined();
```

File: src/statuspage.test.ts (hunk 3 above)
Comment (maveco69) [inline src/statuspage.test.ts]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
import DefaultOperations from "./defaultOperations";
```

File: src/statuspage.ts (hunk 4 above)
Comment (maveco69) [inline src/statuspage.ts]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
import DefaultOperations from "./defaultOperations";
```

File: src/statuspage.test.ts (hunk 6 above)
Comment (maveco69) [inline src/statuspage.test.ts]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
test("I can require StatusPage", () => {
```

File: src/statuspage.test.ts (hunk 5 above)
Comment (maveco69) [inline src/statuspage.test.ts]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
import StatusPage from "./statuspage";
```

File: jest.config.js (hunk 2 above)
Comment (maveco69) [inline jest.config.js]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
  preset: "ts-jest",
```

File: src/statuspage.test.ts (hunk 8 above)
Comment (maveco69) [inline src/statuspage.test.ts]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
test("I can import StatusPage", () => {
```

File: src/statuspage.test.ts (hunk 9 above)
Comment (maveco69) [inline src/statuspage.test.ts]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/zipper-team/statuspage/pullRequest?prid=7096534)
```suggestion
    expect(require("./statuspage")).toBeDefined();
```

File: src/defaultOperations.ts [hunk 10]
```diff
@@ -0,0 +1,45 @@
+import axios from 'axios';
+class DefaultOperations<T> {
```
Comment (jadsonmd) [inline src/defaultOperations.ts]: I rather to another name for this class.
for example: CrudDefaultOperations or just CrudOperations

File: src/defaultOperations.test.ts [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+  expect(defaultOperations.pageId).toBe("PageId");
+  expect(defaultOperations.path).toBe("Path");
+  expect(defaultOperations.config).toStrictEqual({
+    headers: {
+      Authorization: "OAuth ApiKey",
+    },
+  });
+});
+
+test("verify GET", async () => {
```
Comment (rdoVieira01) [inline src/defaultOperations.test.ts]: I feel like the test's names doesn't really say anything.

At the very least it should be capitalized.

**Review by jadsonmd — COMMENTED** [review]
ok

File: src/defaultOperations.ts [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+    this.pageId = pageId;
+    this.config = {
+      headers: {
+        Authorization: `OAuth ${apiKey}`,
+      },
+    };
+  }
+
+  async get(id: String): Promise<T> {
+    let res = await axios.get(
```
Comment (jadsonmd) [inline src/defaultOperations.ts]: I think you should use 'const' instead of 'let' whenever the variable is not reassigned.

Comment (rdoVieira01) [reply src/defaultOperations.ts]: There are some other small things that should appear as a problem with eslint (trailing commas, for example).

**Review by rdoVieira01 — APPROVED** [review]

File: .eslintrc.json [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+        "airbnb-base"
+    ],
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+        "ecmaVersion": 12
+    },
+    "plugins": [
+        "@typescript-eslint"
+    ],
+    "rules": {
```
Comment (brunocordioli072) [inline .eslintrc.json:16]: import/extensions, import/no-unresolved, global-require seems to be useless for the project, they should be turned off.
Like this: 

"rules": {
        "import/extensions": "off",
        "import/no-unresolved": "off",
        "global-require": "off"
    },

**Review by jadsonmd — APPROVED** [review]

File: .eslintrc.json (hunk 13 above)
Comment (maveco69) [reply .eslintrc.json:16]: Agreed on first two, global-require is not really an allowed rule globally.

**Review by brunocordioli072 — APPROVED** [review]

**Pull request merged**